### PR TITLE
Like param issue

### PR DESCRIPTION
**Pull request by Nike682631**
Making a draft pull request for the like flag parameter addition for  dataset functions

### CODE REVIEW

File: packages/cli/src/zosfiles/create/like/like.definition.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        optional: ["zosmf"]
+    },
+    positionals: [
+        {
+            name: "dataSetName",
+            type: "string",
+            description: strings.POSITIONALS.DATASETNAME,
+            required: true
+        },
+        {
```
Comment (awharn) [inline packages/cli/src/zosfiles/create/like/like.definition.ts:40]: Unless my interpretation of the command is incorrect, I do not believe that we want likeDataSetName to be a positional - instead, I believe it should be specified after `--like`, and that is covered by `ZosFilesCreateExtraOptions.like` in the list of options.

File: packages/cli/src/zosfiles/create/like/like.definition.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            name: "likeDataSetName",
+            type: "string",
+            description: strings.POSITIONALS.LIKEDATASETNAME,
+            required: true
+        }
+    ],
+    options: [
```
Comment (awharn) [inline packages/cli/src/zosfiles/create/like/like.definition.ts:47]: For the initial implementation of this command, I believe the only option should be `ZosFilesCreateExtraOptions.like`.

File: packages/cli/src/zosfiles/-strings-/en.ts [hunk 3]
```diff
@@ -135,9 +141,10 @@ export default {
                     SPECIFY_MODE: `Create a USS directory named "testDir" with mode "rwxrwxrwx" `
                 }
             }
-        },
+        }, 
         POSITIONALS: {
-            DATASETNAME: "The name of the data set that you want to create"
+            DATASETNAME: "The name of the data set that you want to create",
+            LIKEDATASETNAME: "Name of the dataset to be liked"
```
Comment (awharn) [inline packages/cli/src/zosfiles/-strings-/en.ts:147]: If the likeDataSetName positional is removed, please remove this string as well.

File: packages/cli/src/zosfiles/-strings-/en.ts [hunk 4]
```diff
@@ -159,7 +167,7 @@ export default {
                 "differs from a partitioned data set (PDS) because it cannot contain members, only file contents.",
             SHOWATTRIBUTES: "Show the full allocation attributes",
             SIZE: "The size of the data set (specified as nCYL or nTRK - where n is the number of cylinders or tracks). Sets the primary " +
-                "allocation (the secondary allocation becomes ~10% of the primary)."
+                "allocation (the secondary allocation becomes ~10% of the primary).",
```
Comment (awharn) [inline packages/cli/src/zosfiles/-strings-/en.ts:170]: Please remove the trailing comma

File: packages/cli/src/zosfiles/create/Create.options.ts [hunk 5]
```diff
@@ -51,6 +51,16 @@ export const ZosFilesCreateExtraOptions: { [key: string]: ICommandOptionDefiniti
         aliases: ["sz"],
         description: strings.SIZE,
         type: "string"
+    },
+        /**
```
Comment (awharn) [inline packages/cli/src/zosfiles/create/Create.options.ts:55]: Please remove the extra spaces so the asterisks align.

File: packages/cli/src/zosfiles/create/like/like.definition.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        ZosFilesCreateOptions.dataclass,
+        ZosFilesCreateOptions.unit,
+        ZosFilesCreateOptions.dsntype,
+        ZosFilesCreateExtraOptions.showAttributes,
+        ZosFilesCreateExtraOptions.like
+    ].sort((a, b) => a.name.localeCompare(b.name)),
+    examples: [
+        {
+            description: strings.ACTIONS.DATA_SET_LIKE.EXAMPLES.EX1,
+            options: "NEW.CLASSIC.DATASET"
```
Comment (awharn) [inline packages/cli/src/zosfiles/create/like/like.definition.ts:67]: This should demonstrate the use of the like option.
```suggestion
            options: "NEW.DATASET --like EXISTING.DATASET"
```

File: packages/cli/src/zosfiles/create/like/like.definition.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { ICommandDefinition } from "@zowe/imperative";
+import { CreateDefaults } from "@zowe/zos-files-for-zowe-sdk";
```
Comment (awharn) [inline packages/cli/src/zosfiles/create/like/like.definition.ts:13]: This import should not be required once below comments are addressed.

File: packages/cli/src/zosfiles/-strings-/en.ts [hunk 8]
```diff
@@ -40,6 +40,12 @@ export default {
                     EX1: "Create an empty C code PDS with default parameters"
                 }
             },
+            DATA_SET_LIKE: {
+                DESCRIPTION: "Create data sets for C code programming",
```
Comment (awharn) [inline packages/cli/src/zosfiles/-strings-/en.ts:44]: This should be something like "Create data sets based on the properties of an existing data set"

**Review by awharn — COMMENTED** [review]
This is coming along nicely. I've left a few suggestions and comments. Please remember to add some unit, integration, and system tests (found in `packages/cli/__tests__/zosfiles`).

Unfortunately, some commits in your pull request do not have DCO signoffs, which are required for this project. Instructions on how to amend your previous commits to provide DCO signoffs can be found under DCO in the Checks tab, or by clicking the link [here](https://github.com/zowe/zowe-cli/pull/894/checks?check_run_id=1656466419). Once you've done that, adding a DCO signoff to a commit should be as easy as appending `--signoff` to the `git commit` command (or, if you're using Visual Studio Code, setting git.alwaysSignOff to `true` in your settings).

Thanks again for making this contribution. It's looking good so far. 😄

**Comment by t1m0thyj** [comment]
Closing this PR in favor of #899 which has the commit history fixed.

**Pull request closed without merging**